### PR TITLE
Overhaul RNW CLI documentation

### PR DESCRIPTION
**Pull request by jonthysell**
## Description

This PR does multiple things in order to improve the RNW CLI documentation, including:

* Creating separate pages for each CLI command
* Updating existing documentation to point to these new pages where appropriate
* Updated Getting Started to use the new `init-windows` command
* Updated other references to the old `react-native-windows-init` command with links to the new command and changed functionality
* Clarified references where we used "React Native Windows CLI" as a generic term when we mean specific commands within the CLI

### Why
To improve out documentation in time for the 0.75 release.

Resolves https://github.com/microsoft/react-native-windows/issues/13394

## Screenshots
N/A

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows-samples/pull/947)

### CODE REVIEW

File: docs/codegen-windows-cli.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+Here are the options that `react-native codegen-windows` takes:
+| Option                | Input Type | Description                                      |
+|-----------------------|------------|--------------------------------------------------|
+| `--logging`           | boolean    | Verbose output logging                           |
+| `--check`             | boolean    | Only check whether any codegen files need to change |
+| `--no-telemetry`      | boolean    | Disables sending telemetry that allows analysis of usage and failures of the react-native-windows CLI |
+| `-h`, `--help`        | boolean    | Display help for command                         |
+
+This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See the below for more details.
```
Comment (TatianaKapos) [inline docs/codegen-windows-cli.md:28]: nit: 
```suggestion
This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See below for more details.
```

File: docs/init-windows-cli.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+| Template | Name |
+|:-:|:--|
+| `cpp-app` | React Native Windows Application (New Arch, C++, Win32, Hermes) |
+| `cpp-lib` | React Native Windows Turbo Module (New Arch, C++) |
+| `old/uwp-cpp-app` | React Native Windows Application (Old Arch, UWP, C++) |
+| `old/uwp-cpp-lib` | React Native Windows Library (Old Arch, UWP, C++) |
+| `old/uwp-cs-app` | React Native Windows Application (Old Arch, UWP, C#)  |
+| `old/uwp-cs-lib` | React Native Windows Library (Old Arch, UWP, C#)  |
+
+This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See the below for more details.
```
Comment (TatianaKapos) [inline docs/init-windows-cli.md:44]: nit:
```suggestion
This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See below for more details.
```

File: docs/react-native-windows-cli.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+## `react-native init-windows`
+
+Initializes a new RNW project from a given template.
+
+See the [react-native init-windows docs](init-windows-cli.md) for detailed usage information.
+
+## `react-native run-windows`
+
+Builds your RNW app and starts it on a connected Windows desktop, emulator or device.
```
Comment (TatianaKapos) [inline docs/react-native-windows-cli.md:30]: nit:
```suggestion
Builds your RNW app and starts it on a connected Windows desktop, emulator, or device.
```

File: docs/run-windows-cli.md [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 
-## `@react-native-windows/cli`
+## `run-windows`
 
-`@react-native-windows/cli` is a CLI used to build and run React Native for Windows apps. 
+The `run-windows` CLI command is used to build and run React Native for Windows apps. 
 
 ### Usage
-Run this from an existing React Native for Windows app to build and deploy.
+Builds your RNW app and starts it on a connected Windows desktop, emulator or device.
```
Comment (TatianaKapos) [inline docs/run-windows-cli.md:13]: nit:
```suggestion
Builds your RNW app and starts it on a connected Windows desktop, emulator, or device.
```

**Review by TatianaKapos — APPROVED** [review]
LGTM! :) some small nits if you want to take them

File: docs/run-windows-cli.md [hunk 5]
```diff
@@ -23,35 +23,33 @@ Here are the options that `react-native run-windows` takes:
...
 | `--buildLogDirectory` | string     | Optional directory where MSBuild log files should be stored |
 | `--info`              | boolean    | Dump environment information                     |
 | `--direct-debugging`  | number     | Enable direct debugging on specified port        |
-| `--useWinUI3`         | boolean    | Targets `WinUI` 3.0 (Preview) instead of UWP XAML.  |
-| `--useHermes`         | boolean    | Use Hermes instead of Chakra as the JS engine (supported on 0.64+)   |
 | `--no-telemetry`      | boolean    | Disables sending telemetry that allows analysis of usage and failures of the react-native-windows CLI |
-| `-h`, `--help`        | boolean    | output usage information                         |
+| `-h`, `--help`        | boolean    | Display help for command                         |
 
 This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See the below for more details.
```
Comment (jonthysell) [inline docs/run-windows-cli.md:50]:
```suggestion
This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See below for more details.
```

File: docs/autolink-windows-cli.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+| Option                | Input Type | Description                                      |
+|-----------------------|------------|--------------------------------------------------|
+| `--logging`           | boolean    | Verbose output logging                           |
+| `--check`             | boolean    | Only check whether any autolinked files need to change |
+| `--sln`               | string     | Override the app solution file determined by 'react-native config', i.e. `windows\myApp.sln` |
+| `--proj`              | string     | Override the app project file determined by 'react-native config', i.e. `windows\myApp\myApp.vcxproj` |
+| `--no-telemetry`      | boolean    | Disables sending telemetry that allows analysis of usage and failures of the react-native-windows CLI |
+| `-h`, `--help`        | boolean    | Display help for command                         |
+
+This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See the below for more details.
```
Comment (jonthysell) [inline docs/autolink-windows-cli.md:32]:
```suggestion
This sends telemetry to Microsoft by default. You can prevent the telemetry from being sent by using the `--no-telemetry` command line option. See below for more details.
```

**Pull request merged**